CONTRIBUTING.md

```diff
@@ -33,15 +37,55 @@ Here are the basic steps to get started with your first contribution. Please rea
 
 See the wiki for more details about our [merging strategy](https://github.com/microsoft/recommenders/wiki/Strategy-to-merge-the-code-to-main-branch).
 
+## Ideas for Contributions
+
+### A first contribution
+
+For people who are new to open source or to Recommenders, a good way to start is by contribution with documentation. You can help with any of the README files or in the notebooks.
+
+For more advanced users, consider fixing one of the bugs listed in the issues.
+
+### Datasets
+
+To contribute new datasets, please consider this:
+
+* Minimize dependencies, it's better to use `requests` library than a custom library.
+* Make sure that the dataset is publicly available and that the license allows for redistribution.
+
+### Models
+
+To contribute new models, please consider this:
+
+* Please don't add models that are already implemented in the repo. An exception to this rule is if you are adding a more optimal implementation or you want to migrate a model from TensorFlow to PyTorch.
+* Prioritize the minimal code necessary instead of adding a full library. If you add code from another repository, please make sure to follow the license and give proper credit.
+* All models should be accompanied by a notebook that shows how to use the model and how to train it. The notebook should be in the [examples](examples) folder.
+* The model should be tested with unit tests, and the notebooks should be tested with functional tests.
+
+### Metrics
+
+To contribute new metrics, please consider this:
+
+* A good way to contribute with metrics is by optimizing the code of the existing ones.
+* If you are adding a new metric, please consider adding not only a CPU version, but also a PySpark version.
+* When adding the tests, make sure you check for the limits. For example, if you add an error metric, check that the error between two identical datasets is zero.
+
+### General tips
+
+* Minimize dependencies. Around 80% of the issues in the repo are related to dependencies.
+* Prioritize PyTorch over TensorFlow.
+* Avoid adding code with GPL and other viral licenses. Prioritize MIT, Apache, and other permissive licenses.
```

Comment:
"viral" is not clear here, not sure what it means.

Reply:
I'll add the copyright note.

About the viral license, it is the name typically given to GPL and other licenses that forces all the code to use that license: https://joinup.ec.europa.eu/collection/eupl/news/why-viral-licensing-ghost

For example, we have MIT, if we add a piece of code with GPL license, that forces us to use the same license for all the code, that is why it is called viral. If instead we add Apache or others, both licenses can be together without a problem. A GPL license, for example, doesn't allow use private code.

There is a lot of controversy with GPL, because they claim they are free, but they force others to do what they want. It is an "interesting freedom" lol.

Reply:
@anargyri updated